sklearn/metrics/_classification.py

```diff
@@ -1955,10 +1993,10 @@ def precision_recall_fscore_support(
         weights = None
 
     if average is not None:
-        assert average != "binary" or len(precision) == 1
-        precision = _nanaverage(precision, weights=weights)
-        recall = _nanaverage(recall, weights=weights)
-        f_score = _nanaverage(f_score, weights=weights)
+        assert average != "binary" or precision.shape[0] == 1
```

Comment:
I'm okay to leave this as is in this PR with this since it's existing code, but we really shouldn't be `assert` ing here. If this never happens, then the line shouldn't be here, if it can happen, we should raise a meaningful error.